cluster/gke/config-test.sh

```diff
@@ -18,11 +18,22 @@
 CLUSTER_NAME="${CLUSTER_NAME:-${USER}-gke-e2e}"
 NETWORK=${KUBE_GKE_NETWORK:-e2e}
 NODE_TAG="k8s-${CLUSTER_NAME}-node"
-IMAGE_TYPE="${KUBE_GKE_IMAGE_TYPE:-container_vm}"
+IMAGE_TYPE="${KUBE_GKE_IMAGE_TYPE:-cos}"
 ENABLE_KUBERNETES_ALPHA="${KUBE_GKE_ENABLE_KUBERNETES_ALPHA:-}"
 
 KUBE_DELETE_NETWORK=${KUBE_DELETE_NETWORK:-true}
 
+# Set e2e test context through NODE_OS_DISTRIBUTION.
+case $(kubectl get node -o=jsonpath='{.items[0].status.nodeInfo.osImage}') in
```

Comment:
I was concerned about this too, but it basically works.
In the `--up` case, the nodes won't have started, so NODE_OS_DISTRIBUTION isn't set.
We [source `config-test.sh` in `hack/ginkgo-e2e.sh`](https://github.com/kubernetes/kubernetes/blob/0744964956abdb17586361dee59853f6167b58d7/hack/ginkgo-e2e.sh#L48), which is where this value actually matters, so we should be OK. (Looking at the build log for the gke and gke-gci runs confirms that things work as intended.)